solana/pyth2wormhole/client/src/main.rs

```diff
@@ -239,66 +256,116 @@ fn handle_attest(
     let config =
         Pyth2WormholeConfig::try_from_slice(rpc.get_account_data(&p2w_config_addr)?.as_slice())?;
 
-    // Derive dynamic seeded accounts
     let seq_addr = Sequence::key(
         &SequenceDerivationData {
             emitter_key: &emitter_addr,
         },
         &config.wh_prog,
     );
 
-    // Arrange Attest accounts
-    let acc_metas = vec![
-        // payer
-        AccountMeta::new(payer.pubkey(), true),
-        // system_program
-        AccountMeta::new_readonly(system_program::id(), false),
-        // config
-        AccountMeta::new_readonly(p2w_config_addr, false),
-        // pyth_product
-        AccountMeta::new_readonly(product_addr, false),
-        // pyth_price
-        AccountMeta::new_readonly(price_addr, false),
-        // clock
-        AccountMeta::new_readonly(clock::id(), false),
-        // wh_prog
-        AccountMeta::new_readonly(config.wh_prog, false),
-        // wh_bridge
-        AccountMeta::new(
-            Bridge::<{ AccountState::Initialized }>::key(None, &config.wh_prog),
-            false,
-        ),
-        // wh_message
-        AccountMeta::new(message_keypair.pubkey(), true),
-        // wh_emitter
-        AccountMeta::new_readonly(emitter_addr, false),
-        // wh_sequence
-        AccountMeta::new(seq_addr, false),
-        // wh_fee_collector
-        AccountMeta::new(FeeCollector::<'_>::key(None, &config.wh_prog), false),
-        AccountMeta::new_readonly(rent::id(), false),
-    ];
-
-    let ix_data = (
-        pyth2wormhole::instruction::Instruction::Attest,
-        AttestData {
-            nonce,
-            consistency_level: ConsistencyLevel::Finalized,
-        },
-    );
-
-    let ix = Instruction::new_with_bytes(p2w_addr, ix_data.try_to_vec()?.as_slice(), acc_metas);
+    // Read the current max batch size from the contract's settings
+    let max_batch_size = config.max_batch_size;
+
+    let mut nonce = start_nonce;
+
+    let mut txs = Vec::with_capacity(attestation_cfg.symbols.len() / max_batch_size as usize);
```

Comment:
I think this needs to round up in case `attestation_cfg.symbols.len()` is not evenly divisible by `max_batch_size` ?

Reply:
This may be premature of me. The pre-allocation may not be worth the raised eyebrows. I would not expect batch counts to go higher  than a thousand or two.. Removing in favor of starting with empty vector.